evennia/web/webclient/static/webclient/js/webclient_gui.js

```diff
@@ -12,618 +12,258 @@
+    var onGotOptions = function (args, kwargs) {
+        Object.keys(plugins).forEach( function (plugin) {
+            if( 'onGotOptions' in plugins[plugin] ) {
+                plugins[plugin].onGotOptions(args, kwargs);
+            }
+        });
     }
-}
 
-var resizeInputField = function () {
-    return function() {
-      var wrapper = $("#inputform")
-      var input = $("#inputcontrol")
-      var prompt = $("#prompt")
 
-      input.height(wrapper.height() - (input.offset().top - wrapper.offset().top));
-    }
-}();
-
-// Handle resizing of client
-function doWindowResize() {
-      resizeInputField();
-      var resizable = $("[data-update-append]");
-      var parents = resizable.closest(".split")
-      parents.animate({
-          scrollTop: parents.prop("scrollHeight")
-      }, 0);
-}
-
-// Handle text coming from the server
-function onText(args, kwargs) {
-    var use_default_pane = true;
-
-    if ( kwargs && 'type' in kwargs ) {
-        var msgtype = kwargs['type'];
-        if ( ! known_types.includes(msgtype) ) {
-            // this is a new output type that can be mapped to panes
-            console.log('detected new output type: ' + msgtype)
-            known_types.push(msgtype);
-        }
+    // Handle text coming from the server
+    var onText = function (args, kwargs) {
+        var use_default_pane = true;
 
-        // pass this message to each pane that has this msgtype mapped
-        if( SplitHandler ) {
-            for ( var key in SplitHandler.split_panes) {
-                var pane = SplitHandler.split_panes[key];
-                // is this message type mapped to this pane?
-                if ( (pane['types'].length > 0) && pane['types'].includes(msgtype) ) {
-                    // yes, so append/replace this pane's inner div with this message
-                    var text_div = $('#'+key+'-sub');
-                    if ( pane['update_method'] == 'replace' ) {
-                        text_div.html(args[0])
-                    } else {
-                        text_div.append(args[0]);
-                        var scrollHeight = text_div.parent().prop("scrollHeight");
-                        text_div.parent().animate({ scrollTop: scrollHeight }, 0);
-                    }
-                    // record sending this message to a pane, no need to update the default div
+        // does this plugin handle this onText event?
+        Object.keys(plugins).forEach( function (plugin) {
+            if( 'onText' in plugins[plugin] ) {
+                if( plugins[plugin].onText(args, kwargs) ) {
+                    // True -- means this plugin claims this text.
+                    // It should not be appended to #messagewindow
+                    // Note: multiple plugins can all claim this text.
                     use_default_pane = false;
                 }
             }
+        });
+
+        // append message to default pane, then scroll so latest is at the bottom.
```

Comment:
This seems to break with the plugin separation used elsewhere in this code. Would it not make more sense to provide this handling as a default plugin-implementation rather than to hard-code it here?